### DISCUSSION TITLE
Reload System Variables After Running Shell Commands

### DISCUSSION THREAD

**Discussion by humbertocarvalho** [dbeaver/dbeaver#12500] [Q&A] [answered]
After running a shell script (e.g Before Connect), I expect the application to refresh the System Variables, because I want to use a batch script to set a variable with my gcloud token. Without this, I have to close the application and open it again.

It would be nice if the application can reload the system variables.

**Comment by p91paul** [comment]
Do you mean environment variables? I'm afraid that's now how env variables work, I might be wrong but they cannot just be "reloaded" I think. Surely on linux, most likely it's the same on windows. How are you setting them?

**Comment by humbertocarvalho** [comment]
> Do you mean environment variables? I'm afraid that's now how env variables work, I might be wrong but they cannot just be "reloaded" I think. Surely on linux, most likely it's the same on windows. How are you setting them?

Yes. I set then through a batch script and refresh windows env. The application can't reload it?

**Comment by p91paul** [comment]
Which command are you using to set them in your batch script?

**Comment by humbertocarvalho** [comment]
I'm using the SETX command.

**Comment by serge-rider** [comment, marked as answer]
I'm afraid it is not possible in Java. You can't "refresh" environment variables, only application restart will work.
